packages/rs-drive-abci/src/logging.rs

```diff
@@ -686,34 +764,69 @@ pub fn init_for_tests(verbosity: u8) {
     logger_builder
         .add("default", &config)
         .expect("cannot configure default logger");
+
     logger_builder.build().try_install().ok();
 }
 
-/// Verify log directory path and determine absolute path to log file.
-fn log_file_path<T: AsRef<Path>>(log_dir: T) -> Result<PathBuf, Error> {
-    let log_dir = log_dir.as_ref();
-    if !log_dir.is_absolute() {
+/// Verify log file path.
+///
+/// Ensure that the log file path is correct, that is:
+/// - it points to a file, not a directory
+/// - if the log file exists, it is writable for the current user
+/// - parent directory of the file exists and is writable for current user
+/// - path is absolute
+fn validate_log_path<T: AsRef<Path>>(log_file_path: T) -> Result<(), Error> {
+    let log_file_path = log_file_path.as_ref();
+
+    if !log_file_path.is_absolute() {
         return Err(Error::FilePath(
-            log_dir.to_owned(),
-            "log path must be absolute".to_string(),
+            log_file_path.to_owned(),
+            "log file path must be absolute".to_string(),
         ));
     }
-    if !log_dir.is_dir() {
+
+    if log_file_path.extension().is_none()
+        || log_file_path.ends_with(String::from(path::MAIN_SEPARATOR))
+    {
         return Err(Error::FilePath(
-            log_dir.to_owned(),
-            "log path must point to an existing directory".to_string(),
+            log_file_path.to_owned(),
+            "log file path must point to file".to_string(),
         ));
     }
-    let path = log_dir.join("drive-abci.log");
-    Ok(path)
+
+    let Some(parent_dir) = log_file_path.parent() else {
```

Comment:
ChatGPT says:

Log file exists but it's read-only: If the log file already exists but it's read-only, the function should return an error. However, this case is not handled by your current implementation.

```
// Assuming /logs/read_only_log.txt is read-only
let result = validate_log_path("/logs/read_only_log.txt");
assert!(result.is_err()); // This assertion will fail with the current implementation
```